src/gui/AstroCalcDialog.cpp

```diff
@@ -4473,6 +4480,19 @@ QMap<double, double> AstroCalcDialog::findClosestApproach(PlanetP& object1, Stel
 	QMap<double, double> separations;
 	QPair<double, double> extremum;
 
+	if (object2->getType()=="Planet")
```

Comment:
Hmm, do you also see the inefficiency?
For a simple date validity test you set the whole program state twice. 

Please do the following: extend
bool Planet::hasValidPositionalData(const double jde)
{
	if (pType<isObserver)
		return true;
	else if (orbitPtr && pType>=isArtificial)
		return static_cast<KeplerOrbit*>(orbitPtr)->objectDateValid(jde);
	else
		return false;
}

And then you can simply test here:

if  ( (!planet->hasValidPositionalData(stopJD)) || (!planet->hasValidPositionalData(startJD)) )
     return separations;  // empty result.

And probably this is not even what we need here? Imagine some object with orbit_good=100 (days)

You make an ephemeris for 250 days which enclose the orbit's validity range. The object would be excluded with the current solution! Maybe add a test for (startJD+stopJD)/2? Or sample every 50 days between startJD and stopJD? Or is there a limit or typical value for (stopJD-startJD)? If this is always something like 10 days, you could exclude the object if both dates are invalid.

Reply:
I've added JDE as parameter to reduce number of calls, but for second proposal I think it should be postponed to the future releases, because this subsystem should be refactored after pencil and paper works.

Reply:
OK. The inefficiency is solved. 
The next question is just what this function should do. Is this logic correct? Whenever at least one limit of the search interval is outside the validity range for the planet, the planet is discarded. I see a danger of another error. If you need to merge this now, leave a FIXME with this question for later, and maybe somebody reports an error. Else please don't merge, and let's solve this with better tests for the overlapping cases. (We may need to add some really simple date validity queries to the Planet.)

Reply:
please add the `QWarning()` logging, or some other warning mechanism. the `QWarning()` is not much asked, and will help if a user opens an issue; the log will explain what happened.

thank you!

Reply:
> The next question is just what this function should do. Is this logic correct? 

Yes, but I'll change it as first iteration of excluding the data.